framework/utils/globalSettings.h

```diff
@@ -48,6 +49,15 @@ namespace Cicada{
             return mResolve;
         }
 
+        void setIpResolveType(int value) {
+            setProperty("IpResolveType", AfString::to_string(value));
```

Comment:
use protected.IpResolveType as key, only the thread that first set it can change the value